server/src/main/java/org/elasticsearch/gateway/PersistedClusterStateService.java

```diff
@@ -1056,15 +1049,31 @@ public void close() throws IOException {
             }
         }
 
-        static class WriterStats {
-            final boolean globalMetaUpdated;
-            final long numIndicesUpdated;
-            final long numIndicesUnchanged;
-
-            WriterStats(boolean globalMetaUpdated, long numIndicesUpdated, long numIndicesUnchanged) {
-                this.globalMetaUpdated = globalMetaUpdated;
-                this.numIndicesUpdated = numIndicesUpdated;
-                this.numIndicesUnchanged = numIndicesUnchanged;
+        static record WriterStats(
```

Comment:
I believe records are static by default, so the `static` modifier seems redundant to me here.

Reply:
++ I opened https://github.com/elastic/elasticsearch/issues/83311.